autoemulate/experimental/emulators/gaussian_process/exact.py

```diff
@@ -121,6 +129,11 @@ def __init__(  # noqa: PLR0913 allow too many arguments since all currently requ
         self.lr = lr
         self.batch_size = batch_size
         self.activation = activation
+        self.gamma = gamma
+        self.optimizer = self.optimizer_cls(self.parameters(), lr=self.lr)  # type: ignore[call-arg] since all optimizers include lr
+        self.scheduler = None
```

Comment:
```suggestion
        self.optimizer = self.optimizer_cls(self.parameters(), lr=self.lr)  # type: ignore[call-arg] since all optimizers include lr
        if scheduler_cls is None:
                self.scheduler = None
        else:
                # here need to also handle whether it's partial or not and do the right thing 
```